gpu/octree/src/octree.cpp

```diff
@@ -131,34 +146,55 @@ void  pcl::gpu::Octree::approxNearestSearchHost(const PointType& query, int& out
 }
                         
 void pcl::gpu::Octree::radiusSearch(const Queries& queries, float radius, int max_results, NeighborIndices& results) const
+{
+    pcl::gpu::Octree::ResultSqrDists result_sqr_distances;
+    radiusSearch(queries, radius, max_results, results, result_sqr_distances);
+}
+
+void pcl::gpu::Octree::radiusSearch(const Queries& queries, const Radiuses& radiuses, int max_results, NeighborIndices& results) const
+{
+    pcl::gpu::Octree::ResultSqrDists result_sqr_distances;
+    radiusSearch(queries, radiuses, max_results, results, result_sqr_distances);
+}
+
+void pcl::gpu::Octree::radiusSearch(const Queries& queries, const Indices& indices, float radius, int max_results, NeighborIndices& results) const
+{
+    pcl::gpu::Octree::ResultSqrDists result_sqr_distances;
+    radiusSearch(queries, indices, radius, max_results, results, result_sqr_distances);
+}
+
+void pcl::gpu::Octree::radiusSearch(const Queries& queries, const float radius, const int max_results, NeighborIndices& results, ResultSqrDists& sqr_distances) const
 {
     assert(queries.size() > 0);
     results.create(static_cast<int> (queries.size()), max_results);
     results.sizes.create(queries.size());
-    
+    sqr_distances.create(queries.size() * max_results);
```

Comment:
> pretty sure there's a  better (and probably obvious :stuck_out_tongue:  ) method for this that i'm missing.

Yeah. You need to move all those allocations to `pcl::device::OctreeImpl::radiusSearchEx`, which is always called with all arguments.

```cpp
template<typename BatchType>
void pcl::device::OctreeImpl::radiusSearchEx(BatchType& batch, const Queries& queries, NeighborIndices& results, BatchResultSqrDists& sqr_distances, const bool compute_all_distances)
{
    // no longer spread in all overloads
    assert(queries.size() > 0);
    results.create(static_cast<int> (queries.size()), max_results);
    results.sizes.create(queries.size());
    sqr_distances.create(queries.size() * max_results);


    batch.indices = indices;
    batch.octree = octreeGlobal;

    batch.compute_all_distances = compute_all_distances;
    batch.max_results = results.max_elems;
    batch.output = results.data;                
    batch.output_sizes = results.sizes;
    batch.sqr_distances = sqr_distances;

    batch.points = points_sorted;
    [...]
```